catalog/content/core/content-core-catalogerplugin/src/test/java/ddf/content/plugin/cataloger/TestCatalogContentPlugin.java

```diff
@@ -0,0 +1,498 @@
+import static org.mockito.Matchers.eq;
+import static org.mockito.Matchers.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.activation.MimeType;
+
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.session.mgt.SimpleSession;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.util.ThreadContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.dynamic.api.DynamicMetacard;
+import ddf.catalog.data.dynamic.api.MetacardFactory;
+import ddf.catalog.data.dynamic.impl.MetacardFactoryImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.impl.UpdateImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.content.data.ContentItem;
+import ddf.content.operation.CreateRequest;
+import ddf.content.operation.CreateResponse;
+import ddf.content.operation.DeleteResponse;
+import ddf.content.operation.UpdateRequest;
+import ddf.content.operation.UpdateResponse;
+import ddf.content.plugin.ContentPlugin;
+import ddf.content.plugin.PluginExecutionException;
+import ddf.mime.MimeTypeToTransformerMapper;
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+
+public class TestCatalogContentPlugin {
+
+    private static final String DEFAULT_METACARD_TRANSFORMER = "geojson";
+    private static final String MIME_BASE = "text";
+    private static final String MIME_SUB = "plain";
+    private static final String MIME_TYPE = MIME_BASE + "/" + MIME_SUB;
+    private static final String TEST_USER_NAME = "TestUser";
+    private static final String TEST_REALM = "testrealm";
+    private static final String TEST_DATA = "Test String From InputStream";
+    private static final String TEST_URI = "http://localhost/test.jpg";
+    private static final String BAD_URI = "bad uri";
+
+    private CatalogFramework mockCatalogFramework = mock(CatalogFramework.class);
+    private MimeTypeToTransformerMapper mockMimeTypeToTransformerMapper = mock(MimeTypeToTransformerMapper.class);
+    private BinaryContent mockBinaryContent = mock(BinaryContent.class);
+    private CreateRequest mockCreateRequest = mock(CreateRequest.class);
+    private UpdateRequest mockUpdateRequest = mock(UpdateRequest.class);
+    private CreateResponse mockInputCreateResponse = mock(CreateResponse.class);
```

Comment:
Maybe these should be named `mockContentCreateResponse`, etc., and the mock catalog responses named `mockCatalogCreateResponse`, etc.?